nav2_regulated_pure_pursuit_controller/src/path_handler.cpp

```diff
@@ -47,12 +47,17 @@ double PathHandler::getCostmapMaxExtent() const
 
 nav_msgs::msg::Path PathHandler::transformGlobalPlan(
   const geometry_msgs::msg::PoseStamped & pose,
-  double max_robot_pose_search_dist)
+  double max_robot_pose_search_dist,
+  bool reject_unit_path)
 {
   if (global_plan_.poses.empty()) {
     throw nav2_core::InvalidPath("Received plan with zero length");
   }
 
+  if (reject_unit_path && global_plan_.poses.size() == 1) {
```

Comment:
when is `reject_unit_path` ever true?

Reply:
When `params_->interpolate_curvature_after_goal` is `true`
https://github.com/doisyg/navigation2/blob/994351e100c50f23c3886cf2375bd342e0fbad9f/nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp#L179-L181